src/Core/FlatpakBackend.vala

```diff
@@ -73,6 +75,23 @@ public class AppCenterCore.FlatpakBackend : Backend, Object {
         appstream_pool.set_cache_flags (AppStream.CacheFlags.NONE);
         package_list = new Gee.HashMap<string, Package> (null, null);
 
+        // Monitor the FlatpakInstallation for changes (e.g. adding/removing remotes)
+        if (installation != null) {
+            try {
+                installation_changed_monitor = installation.create_monitor ();
+            } catch (Error e) {
+                critical ("Couldn't create Installation File Monitor : %s", e.message);
+            }
+
+            installation_changed_monitor.changed.connect (() => {
+                debug ("Flatpak installation changed.");
+                var cache_cancellable = new GLib.Cancellable ();
+                refresh_cache (cache_cancellable);
+            });
+        } else {
+            error ("Couldn't create Installation File Monitor due to no installation");
```

Comment:
Best not to use `error` in GLib code as it's fatal for the program, use `critical` or `warning` instead. Warning is probably most appropriate for any of the messages relating to the monitor as it doesn't stop appcenter from working completely if it's broken.